dev/make/cmplr.gnu.ref.mk

```diff
@@ -0,0 +1,51 @@
+#===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+#++
+#  g++ defenitions for makefile
+#--
+
+PLATs.gnu = lnx32e mac32e
+
+CMPLRDIRSUFF.gnu = _gnu
+
+CORE.SERV.COMPILER.gnu = generic
+
+-Zl.gnu =
+-DEBC.gnu = -g
+
+COMPILER.all.gnu =  ${CXX} -m64 -fwrapv -fno-strict-overflow -fno-delete-null-pointer-checks \
+                    -DDAAL_REF -Werror -Wreturn-type
```

Comment:
The source code uses both DAAL_REF and ONEDAL_REF. Changing all of these to be the same (either all DAAL_REF or all ONEDAL_REF) means that undefined symbol errors are not encountered.

As it stands, some of the avx512, avx2, etc. functions from micromkl get referenced from the code base

Reply:
As an example of difference:
#ifdef ONEDAL_REF
#if (defined(DAAL_REF) && (INTPTR_MAX == INT64_MAX))
using GEMM_INT = std::int64_t;
#else
using GEMM_INT = std::int32_t;
#endif